pym/bob/cmds/build.py

```diff
@@ -611,12 +623,20 @@ def _cookCheckoutStep(self, checkoutStep, done, depth):
                         .format(prettySrcPath), "33"))
                 else:
                     self._info("   CHECKOUT  skipped due to --build-only ({})".format(prettySrcPath))
-            elif (self.__force or (not checkoutStep.isDeterministic()) or
+            elif ((self.__force or (not checkoutStep.isDeterministic()) or
```

Comment:
I think the logic here and below has to be changed. The status of a directory must only be checked if the scmDigest of `oldCheckoutState` matches the one in `checkoutState`. Otherwise the directory content is not known. Anything else will be moved away anyway. All in all the logic should look like this unless I forgot something:

```python
if self.__buildOnly and (BobState().getResultHash(prettySrcPath) is not None):
  ...
else:
  if self.__forceCheckout:
    # check state of SCMs and invalidate if the directory is unclean
    stats = {}
    for scm in checkoutStep.getScmList():
      stats.update({ dir : scm for dir in scm.getDirectories.keys() })
    for (scmDir, scmDigest) in oldCheckoutState.copy().items():
      if scmDir is None: continue
      if scmDigest != checkoutState.get(scmDir): continue
      if stats[scmDir].status(checkoutStep.getWorkspacePath(), scmDir, False) == 'unclean':
        oldCheckoutState[scmDir] = None
  if (self.__force or (not checkoutStep.isDeterministic()) or
    (BobState().getResultHash(prettySrcPath) is None) or
    (checkoutState != oldCheckoutState)):
    ...
  else:
    self._info("   CHECKOUT  skipped (fixed package {})".format(prettySrcPath))
```